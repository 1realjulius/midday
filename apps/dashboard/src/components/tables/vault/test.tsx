"use client";

import { useVaultContext } from "@/store/vault/hook";
import { Button } from "@midday/ui/button";

export function Test() {
  const createFolder = useVaultContext((s) => s.createFolder);

  const handleCreateFolder = () => {
    createFolder({ name: "Untitled folder" });
  };

  return (
    <Button onClick={handleCreateFolder} variant="outline">
      Create folder
    </Button>
  );
}
