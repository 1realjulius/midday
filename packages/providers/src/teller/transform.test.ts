import { expect, test } from "bun:test";

const transformedSnapshot = {
  date: "2024-03-04",
  name: "Cash Deposit",
  method: "deposit",
  internal_id: "123_txn_os41r5u90e29shubl2001",
  amount: "57.43",
  currency: "USD",
  bank_account_id: "a87e60b9-8323-4bab-8dc9-4ac9786b48a6",
  category: "income",
  team_id: "123",
  balance: "83353.83",
  status: "posted",
};

test("2 + 2", () => {
  expect(2 + 2).toBe(4);
});
