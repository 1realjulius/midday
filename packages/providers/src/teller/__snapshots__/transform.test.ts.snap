// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Transform pending transaction 1`] = `
{
  "amount": 83.62,
  "balance": null,
  "bank_account_id": "123",
  "category": "income",
  "currency": "USD",
  "date": "2024-03-05",
  "internal_id": "123_txn_os41r5u90e29shubl2000",
  "method": "other",
  "name": "Online Check Deposit",
  "status": "pending",
  "team_id": "123",
}
`;

exports[`Transform card payment transaction 1`] = `
{
  "amount": -68.9,
  "balance": "83431.46",
  "bank_account_id": "123",
  "category": "uncategorized",
  "currency": "USD",
  "date": "2024-03-01",
  "internal_id": "123_txn_os41r5u90e29shubl2005",
  "method": "card_purchase",
  "name": "Nordstrom",
  "status": "posted",
  "team_id": "123",
}
`;

exports[`Transform income transaction 1`] = `
{
  "amount": -20.21,
  "balance": "83296.40",
  "bank_account_id": "123",
  "category": "uncategorized",
  "currency": "USD",
  "date": "2024-03-03",
  "internal_id": "123_txn_os41r5u90e29shubl2002",
  "method": "card_purchase",
  "name": "Exxon Mobil",
  "status": "posted",
  "team_id": "123",
}
`;

exports[`Transform type transfer 1`] = `
{
  "amount": -37.99,
  "balance": "85897.25",
  "bank_account_id": "123",
  "category": "transfer",
  "currency": "USD",
  "date": "2024-01-27",
  "internal_id": "123_txn_os41r5ua0e29shubl2001",
  "method": "transfer",
  "name": "Recurring Transfer To Savings",
  "status": "posted",
  "team_id": "123",
}
`;
